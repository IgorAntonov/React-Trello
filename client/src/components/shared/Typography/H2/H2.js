import styled from 'styled-components';

import { media } from 'Src/helpers';

export const H2 = styled.h2`
  margin-bottom: 0.5rem;
  margin-top: 0;
  
  font-size: 1.2rem;
  font-weight: 400;

  ${media.mobile`
    font-size: 1.3rem;
  `}
  ${media.tablet`
    font-size: 1.4rem;
    font-weight: 500;
  `}
  ${media.desktop`
    font-size: 1.5rem;
  `}
`;
