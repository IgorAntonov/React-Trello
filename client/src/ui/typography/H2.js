import styled from 'styled-components';
import PropTypes from 'prop-types';

import { media } from '../media-template';

export const H2 = styled.h2`
  margin-top: 0;
  margin-bottom: ${({ marginB }) => marginB || '0.5rem'};
  
  color: ${({ color }) => color || '#fafafa'};
  background-color: ${({ bgcolor }) => bgcolor || 'transparent'};
  
  cursor: default;
  font-size: 1.2rem;
  font-weight: 400;
  user-select: none;

  ${media.mobile`
    font-size: 1.3rem;
  `}
  ${media.tablet`
    font-size: 1.4rem;
    font-weight: 500;
  `}
  ${media.desktop`
    font-size: 1.5rem;
  `}
`;

H2.propTypes = {
  marginB: PropTypes.string,
  color: PropTypes.string,
  bgcolor: PropTypes.string
};
