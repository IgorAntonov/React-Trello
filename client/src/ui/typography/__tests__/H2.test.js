import React from 'react';
import { shallow } from 'enzyme';

import { H2 } from '../H2';

describe('<H2/>', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<H2 />);
  });

  test('renders', () => {
    expect(wrapper.exists()).toBe(true);
  });
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
