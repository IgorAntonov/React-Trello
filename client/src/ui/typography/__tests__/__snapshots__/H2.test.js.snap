// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H2/> should match snapshot 1`] = `
<h2
  className="H2-eUbBFj jzrrlj"
/>
`;
